src/vinyldns/batch_change.py

```diff
@@ -0,0 +1,199 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TODO: Add module docstring."""
+
+from vinyldns.serdes import parse_datetime, map_option
+
+
+class AddRecord(object):
+    def __init__(self, input_name, type, ttl, record):
+        self.input_name = input_name
+        self.type = type
+        self.ttl = ttl
+        self.record = record
+        self.change_type = 'Add'
+
+    @staticmethod
+    def from_dict(d):
+        return AddRecord(
+            input_name=d['inputName'],
+            type=d['type'],
+            ttl=d['ttl'],
+            record=d['record']
+        )
+
+
+class DeleteRecordSet(object):
+    def __init__(self, input_name, type):
+        self.input_name = input_name
+        self.type = type
+        self.change_type = 'DeleteRecordSet'
+
+    @staticmethod
+    def from_dict(d):
+        return DeleteRecordSet(
+            input_name=d['inputName'],
+            type=d['type']
+        )
+
+
+class BatchChangeRequest(object):
+    change_type_converters = {
+        'Add': AddRecord.from_dict,
+        'DeleteRecordSet': DeleteRecordSet.from_dict
+    }
+
+    def __init__(self, changes, comments=None):
+        self.comments = comments
+        self.changes = changes
+
+    @staticmethod
+    def from_dict(d):
+        return BatchChangeRequest(
+            comments=d.get('comments'),
+            changes=[BatchChangeRequest.change_type_converters[elem['changeType']](elem)
+                     for elem in d.get('changes', [])]
+        )
+
+
+class AddRecordChange(object):
+    def __init__(self, zone_id, zone_name, record_name, input_name, type, ttl, record_data, status, id,
+                 system_message=None, record_change_id=None, record_set_id=None):
+        self.zone_id = zone_id
+        self.zone_name = zone_name
+        self.record_name = record_name
+        self.input_name = input_name
+        self.type = type
+        self.ttl = ttl
+        self.record_data = record_data,
+        self.status = status
+        self.id = id
+        self.system_message = system_message
+        self.record_change_id = record_change_id
+        self.record_set_id = record_set_id
+        self.change_type = 'Add'
+
+    @staticmethod
+    def from_dict(d):
+        return AddRecordChange(
+            zone_id=d['zoneId'],
+            zone_name=d['zoneName'],
+            record_name=d['recordName'],
+            input_name=d['inputName'],
+            type=d['type'],
+            ttl=d['ttl'],
+            record_data=d['recordData'],
+            status=d['status'],
+            id=d['id'],
+            system_message=d.get('systemMessage'),
```

Comment:
this distinction is important... `d.get` will return a `None` if the name is not present in the dictionary; whereas, `d['xxx']` will raise an error if the name is not present.  Use the former when a field is optional, or there is a recommended default; otherwise use the latter to raise an error to the client/user in the event that the expected field is not present